staging/src/k8s.io/apiserver/pkg/audit/context.go

```diff
@@ -154,15 +327,18 @@ func WithAuditID(ctx context.Context, auditID types.UID) {
 		return
 	}
 	if ac := AuditContextFrom(ctx); ac != nil {
-		ac.Event.AuditID = auditID
+		ac.auditID = auditID
```

Comment:
is there much benefit to assigning this outside a lock when we still take a lock to set event.AuditID below?

Reply:
happy to move it inside!